packages/features/data-table/lib/server.ts

```diff
@@ -171,3 +176,56 @@ export function makeWhereClause(props: MakeWhereClauseProps) {
   }
   throw new Error(`Invalid filter type: ${JSON.stringify({ columnName, filterValue })}`);
 }
+
+/**
+ * Builds a SQL where clause for use with raw SQL queries
+ */
+export function makeSqlCondition(filterValue: FilterValue): Prisma.Sql | null {
+  if (isMultiSelectFilterValue(filterValue)) {
+    return Prisma.sql`= ANY(${filterValue.data})`;
+  } else if (isSingleSelectFilterValue(filterValue)) {
+    return Prisma.sql`= ${filterValue.data}`;
+  } else if (isTextFilterValue(filterValue)) {
+    const { operator, operand } = filterValue.data;
+    switch (operator) {
+      case "equals":
+        return Prisma.sql`= ${operand}`;
+      case "notEquals":
+        return Prisma.sql`!= ${operand}`;
+      case "contains":
+        return Prisma.sql`ILIKE ${`%${operand}%`}`;
+      case "notContains":
+        return Prisma.sql`NOT ILIKE ${`%${operand}%`}`;
+      case "startsWith":
+        return Prisma.sql`ILIKE ${`${operand}%`}`;
+      case "endsWith":
+        return Prisma.sql`ILIKE ${`%${operand}`}`;
+      case "isEmpty":
+        return Prisma.sql`= ''`;
+      case "isNotEmpty":
+        return Prisma.sql`!= ''`;
+      default:
+        return null;
+    }
+  } else if (isNumberFilterValue(filterValue)) {
+    const { operator, operand } = filterValue.data;
+    switch (operator) {
+      case "eq":
+        return Prisma.sql`= ${operand}`;
+      case "neq":
+        return Prisma.sql`!= ${operand}`;
+      case "gt":
+        return Prisma.sql`> ${operand}`;
+      case "gte":
+        return Prisma.sql`>= ${operand}`;
+      case "lt":
+        return Prisma.sql`< ${operand}`;
+      case "lte":
+        return Prisma.sql`<= ${operand}`;
+      default:
+        return null;
+    }
+  }
+
+  return null;
```

Comment:
Should this throw? Seems weird to be returning an empty query here

Reply:
where we use it, we check if it's truthy first.

<img width="966" height="1159" alt="Screenshot 2025-07-16 at 15 02 41" src="https://github.com/user-attachments/assets/866f96e1-8c88-4bcb-a572-bc3b473072ab" />